src/semaphore_internal.h

```diff
@@ -29,6 +29,13 @@
 
 struct dispatch_queue_s;
 
+#if USE_FUTEX_SEM
+typedef struct {
+	/* Low 32 bits are the count; high bits are the waiter count. */
+	volatile uint64_t dfx_data;
+} dispatch_futex_t;
+#endif  /* USE_FUTEX_SEM */
```

Comment:
I tried to code this up ... this makes the code <ugly> IMO
and also not clear from a semantics point of view, since the code needs to massage the full uint64.
For example:  we compare and exchange of the full value, which believe we have to or NOT?
since data has to be volatile (even for the dfx_futex etc), its not obvious that the compiler won't reload 
it though the same union is accessed.

I suggest not to go this route at this time and leave as is. The code is limited to a few lines.
We can discuss later, but want to get this out and done.

DISPATCH_INLINE
bool
_dispatch_futex_trywait(dispatch_futex_t dfx) 
{
    uint64_t orig;
    while ((orig = dfx->dfx_data) && dfx->dfx_futex) {
        if (dispatch_atomic_cmpxchg2o(dfx, dfx_data, orig, orig - 1, seq_cst)) {
            return true;
        }
    }    
    return false;
}